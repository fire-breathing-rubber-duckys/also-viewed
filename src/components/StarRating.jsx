/* eslint-disable react/no-array-index-key */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const StarRating = ({ rating, ratingCount }) => {
  const [starsArr, setStarsArr] = useState([]);
  useEffect(() => {
    switch (rating) {
      case 0: setStarsArr([0, 0, 0, 0, 0]); break;
      case 0.5: setStarsArr([0.5, 0, 0, 0, 0]); break;
      case 1: setStarsArr([1, 0, 0, 0, 0]); break;
      case 1.5: setStarsArr([1, 0.5, 0, 0, 0]); break;
      case 2: setStarsArr([1, 1, 0, 0, 0]); break;
      case 2.5: setStarsArr([1, 1, 0.5, 0, 0]); break;
      case 3: setStarsArr([1, 1, 1, 0, 0]); break;
      case 3.5: setStarsArr([1, 1, 1, 0.5, 0]); break;
      case 4: setStarsArr([1, 1, 1, 1, 0]); break;
      case 4.5: setStarsArr([1, 1, 1, 1, 0.5]); break;
      case 5: setStarsArr([1, 1, 1, 1, 1]); break;
      default: break;
    }
  }, []);

  return (
    <div>
      {starsArr.map((star, i) => {
        if (star === 0) {
          return <i className="far fa-star" key={i} id="star" />;
        }
        if (star === 0.5) {
          return <i className="fas fa-star-half-alt" key={i} id="star" />;
        }
        if (star === 1) {
          return <i className="fas fa-star" key={i} id="star" />;
        }
        return undefined;
      })}
      <label className="item-rating-count">{ratingCount}</label>
    </div>
  );
};

StarRating.propTypes = {
  rating: PropTypes.number.isRequired,
  ratingCount: PropTypes.string.isRequired,
};

export default StarRating;
