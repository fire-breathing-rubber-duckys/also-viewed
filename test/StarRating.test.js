import jsdom from 'jsdom';
const { JSDOM } = jsdom;
import React from 'react';
import { shallow, mount } from 'enzyme';
import { expect as expectChai } from 'chai';
import renderer from 'react-test-renderer';
import StarRating from '../src/components/StarRating';

const { document } = (new JSDOM('')).window;
global.document = document;
global.window = document.defaultView;

const rating = 4;
const ratingCount = '13';

it('Should render correctly', () => {
  const tree = renderer
    .create(<StarRating rating={rating} ratingCount={ratingCount} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Star Rating Rendering', () => {
  const wrapper = mount(<StarRating rating={rating} ratingCount={ratingCount} />);
  it('Should render five stars', () => {
    expectChai(wrapper.find('#star')).to.be.lengthOf(5);
  });
  it('Should render the correct ratingCount', () => {
    const priceWrapper = wrapper.find('.item-rating-count');
    expectChai(priceWrapper.text()).to.equal(ratingCount);
  });
});
