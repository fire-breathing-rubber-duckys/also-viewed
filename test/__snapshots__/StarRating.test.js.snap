// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div>
  <label
    className="item-rating-count"
  >
    13
  </label>
</div>
`;
